shared/proof.go

```diff
@@ -11,14 +13,10 @@ type ProofMetadata struct {
 
 	Challenge     Challenge
 	NumUnits      uint32
-	BitsPerLabel  uint8
 	LabelsPerUnit uint64
 
 	K1 uint32 // K1 specifies the difficulty for a label to be a candidate for a proof.
 	K2 uint32 // K2 is the number of labels below the required difficulty required for a proof.
```

Comment:
Does `K3` need to be added to this?

Reply:
But a proof is valid if the `K3` labels checked are valid, not if all labels are valid. So different `K3` values can cause a proof to be valid or not...

Reply:
`K3` is not used for proof generation at all. It's a parameter solely for a verifier. The proof is not created "with some K3". K3 could probably even change long after proof had been generated and it should still be valid (unless the prover had ground and had found some subset of labels that satisfy a particular value of K3).